Build/scripts/add_msbuild_path.cmd

```diff
@@ -10,22 +11,48 @@
 set FORCE_MSBUILD_VERSION=%1
 
 if "%FORCE_MSBUILD_VERSION%" == "msbuild14" (
-    echo Skipping Dev16 and trying Dev14...
+    echo Skipping Dev17 and trying Dev14...
     goto :LABEL_USE_MSBUILD_14
 )
 if "%FORCE_MSBUILD_VERSION%" == "msbuild15" (
-    echo Skipping Dev16 and trying Dev15...
+    echo Skipping Dev17 and trying Dev15...
     goto :LABEL_USE_MSBUILD_15
 )
+if "%FORCE_MSBUILD_VERSION%" == "msbuild16" (
+    echo Skipping Dev17 and trying Dev16...
+    goto :LABEL_USE_MSBUILD_16
+)
 
 where /q msbuild.exe
 if "%ERRORLEVEL%" == "0" (
     goto :SkipMsBuildSetup
 )
 
-REM Try Dev16 first
+REM Try Dev17 first, then older versions
+
+echo Trying to locate Dev17...
+
+:LABEL_USE_MSBUILD_17
+set MSBUILD_VERSION=17.0
+set "MSBUILD_PATH=%ProgramFiles(x86)%\Microsoft Visual Studio\Preview\Enterprise\MSBuild\%MSBUILD_VERSION%\Bin"
+
+if not exist "%MSBUILD_PATH%\msbuild.exe" (
+    set "MSBUILD_PATH=%ProgramFiles%\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\amd64"

```

Comment:
This is what does the trick for 64-bit installation, I am not sure if the rest is necessary, don't know if CI would ever have a "Preview" version or 32-bit version installed on 64-bit OS (that is the only way to get `ProgramFiles(x86)`).

Reply:
I got this block by copying the block below, basically, and I have no idea why that tests for so many different combinations. Maybe it is OK to prune all of this, because this is likely to be only use from CI (and especially since it refers to paid Enterprise version of VS).

Reply:
I think, basically it's trying every feasible search path to find msbuild - it sets one then if that's not a valid path it tries the next and so on.

Reply:
Dev16 checks were added in bulk, it's hard to say if we need all of this for dev17. For example, the paths that have `Program Files(x86)` are only going to be valid in a 32-bit installations on 64-bit systems. Another point is this isn't really exhaustive, as 'Community Edition' would have a different paths, this script does check those.